beacon-chain/types/block.go

```diff
@@ -261,6 +275,12 @@ func (b *Block) isAttestationValid(attestationIndex int, db beaconDB, aState *Ac
 	return true
 }
 
+// isRandaoValid verifies the validity of randao from block by comparing it with proposer's randao
+// from crystallized state.
+func (b *Block) isRandaoValid(cStateRandao []byte) bool {
+	return b.RandaoReveal() == hashutil.Hash(cStateRandao)
```

Comment:
I think this is flipped.

The reveal should hash to the commitment found in the crystallized state.

Reply:
Some of your tests will have to change based on this change.

Note that the validator needs to pre-commit to a hash-onion of very high depth of a seed -- `hash(hash(...hash(seed)...))`. when they do a reveal, they are revealing the layer of the onion one layer in from the previous commitment.

The validator will need to remember this pre-committed seed at the very least. They might choose to remember all of the layers of the commitment to be ready on hand.

Reply:
Ha! That's right. So happy to have you audit my code 😅